GCC 13 build failure

Build failure under GCC 13:
```
[1/76] /usr/bin/g++ -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_SHARED_LIB -Dmatrix_client_EXPORTS -I/builddir/build/BUILD/mtxclient-0.9.1/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -U_FORTIFY_SOURCE -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer 		-Wall 		-Wextra 		-Wconversion 		-pipe 		-pedantic 		-fsized-deallocation 		-fdiagnostics-color=always 		-Wno-attributes 		-Wunreachable-code -DNDEBUG -fPIC -pthread -DSPDLOG_SHARED_LIB -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -std=c++20 -MD -MT CMakeFiles/matrix_client.dir/lib/crypto/encoding.cpp.o -MF CMakeFiles/matrix_client.dir/lib/crypto/encoding.cpp.o.d -o CMakeFiles/matrix_client.dir/lib/crypto/encoding.cpp.o -c /builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp
FAILED: CMakeFiles/matrix_client.dir/lib/crypto/encoding.cpp.o 
/usr/bin/g++ -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_SHARED_LIB -Dmatrix_client_EXPORTS -I/builddir/build/BUILD/mtxclient-0.9.1/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -U_FORTIFY_SOURCE -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer 		-Wall 		-Wextra 		-Wconversion 		-pipe 		-pedantic 		-fsized-deallocation 		-fdiagnostics-color=always 		-Wno-attributes 		-Wunreachable-code -DNDEBUG -fPIC -pthread -DSPDLOG_SHARED_LIB -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -std=c++20 -MD -MT CMakeFiles/matrix_client.dir/lib/crypto/encoding.cpp.o -MF CMakeFiles/matrix_client.dir/lib/crypto/encoding.cpp.o.d -o CMakeFiles/matrix_client.dir/lib/crypto/encoding.cpp.o -c /builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp:33:22: error: 'uint8_t' was not declared in this scope
   33 | constexpr std::array<uint8_t, 256>
      |                      ^~~~~~~
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp:6:1: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
    5 | #include <vector>
  +++ |+#include <cstdint>
    6 | 
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp:33:34: error: template argument 1 is invalid
   33 | constexpr std::array<uint8_t, 256>
      |                                  ^
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp:33:16: error: '<expression error>' in namespace 'std' does not name a type
   33 | constexpr std::array<uint8_t, 256>
      |                ^~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp:48:59: error: 'invert_alphabet' was not declared in this scope; did you mean 'base64_alphabet'?
   48 | static constexpr const std::array base64_to_int         = invert_alphabet(base64_alphabet);
      |                                                           ^~~~~~~~~~~~~~~
      |                                                           base64_alphabet
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp:49:59: error: 'invert_alphabet' was not declared in this scope; did you mean 'base64_alphabet'?
   49 | static constexpr const std::array base64_urlsafe_to_int = invert_alphabet(base64_urlsafe_alphabet);
      |                                                           ^~~~~~~~~~~~~~~
      |                                                           base64_alphabet
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp:50:59: error: 'invert_alphabet' was not declared in this scope; did you mean 'base64_alphabet'?
   50 | static constexpr const std::array base58_to_int         = invert_alphabet(base58_alphabet);
      |                                                           ^~~~~~~~~~~~~~~
      |                                                           base64_alphabet
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp: In function 'std::string {anonymous}::encode_base58(const std::array<char, 58>&, const std::string&)':
/builddir/build/BUILD/mtxclient-0.9.1/lib/crypto/encoding.cpp:64:17: error: 'uint8_t' was not declared in this scope
```

Fix:
```diff
#include <vector>
+#include <cstdint>
```

--- a/examples/crypto_bot.cpp
+++ b/examples/crypto_bot.cpp
@@ -5,6 +5,7 @@
 #include "spdlog/spdlog.h"
 
 #include <atomic>
+#include <cstdint>
 #include <fstream>
 #include <iostream>
 #include <nlohmann/json.hpp>

--- a/include/mtx/events/mscs/image_packs.hpp
+++ b/include/mtx/events/mscs/image_packs.hpp
@@ -4,6 +4,7 @@
 /// @brief Image packs from [MSC2545](https://github.com/matrix-org/matrix-doc/pull/2545)
 
 #include <bitset>
+#include <cstdint>
 #include <map>
 #include <string>
 

--- a/include/mtxclient/crypto/objects.hpp
+++ b/include/mtxclient/crypto/objects.hpp
@@ -5,6 +5,7 @@
 ///
 /// The wrappers implement RAII semantics, so you don't need to free stuff manually.
 
+#include <cstdint>
 #include <memory>
 #include <olm/olm.h>
 #include <olm/pk.h>

--- a/include/mtxclient/crypto/utils.hpp
+++ b/include/mtxclient/crypto/utils.hpp
@@ -3,6 +3,7 @@
 /// @file
 /// @brief Various crypto functions.
 
+#include <cstdint>
 #include <string>
 #include <vector>
 

--- a/include/mtxclient/utils.hpp
+++ b/include/mtxclient/utils.hpp
@@ -3,6 +3,7 @@
 /// @file
 /// @brief Various utility functions for http requests.
 
+#include <cstdint>
 #include <iosfwd>
 #include <map>
 #include <string>

--- a/lib/crypto/client.cpp
+++ b/lib/crypto/client.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include <nlohmann/json.hpp>
 #include <utility>
 

--- a/lib/crypto/encoding.cpp
+++ b/lib/crypto/encoding.cpp
@@ -1,6 +1,7 @@
 #include <algorithm>
 #include <array>
 #include <cassert>
+#include <cstdint>
 #include <string>
 #include <vector>
 

--- a/lib/crypto/utils.cpp
+++ b/lib/crypto/utils.cpp
@@ -12,6 +12,7 @@
 #include <olm/pk.h>
 
 #include <algorithm>
+#include <cstdint>
 
 #include "mtx/log.hpp"
 #include "mtxclient/crypto/client.hpp"

--- a/lib/structs/events/encrypted.cpp
+++ b/lib/structs/events/encrypted.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include <string>
 
 #include <nlohmann/json.hpp>

--- a/lib/utils.cpp
+++ b/lib/utils.cpp
@@ -1,6 +1,7 @@
 #include "mtxclient/utils.hpp"
 
 #include <algorithm>
+#include <cstdint>
 #include <iomanip>
 #include <random>
 #include <sstream>

